src/controllers/company.number.controller.ts

```diff
@@ -14,20 +16,27 @@ import logger from "../utils/logger";
  * @param next
  */
 const route = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
-    logger.debugRequest(req, "Attempt to retrieve and show the company details");
+  logger.debugRequest(req, "Attempt to retrieve and show the company details");
 
-    const companyNumber: string = req.body.companyNumber;
-    const session: Session = req.session as Session;
-    logger.infoRequest(req, `Retrieving company profile for company number ${companyNumber}`);
-    const token: string = getValidAccessToken(session) as string;
+  const companyNumber: string = req.body.companyNumber;
+  const session: Session = req.session as Session;
+  logger.infoRequest(req, `Retrieving company profile for company number ${companyNumber}`);
 
-    const company: ObjectionCompanyProfile = await getCompanyProfile(companyNumber, token);
+  let token: string;
+  try {
+    token = retrieveAccessTokenFromSession(session);
+  } catch (e) {
+    return next(e);
+  }
 
-    if (session) {
-        createObjectionsSession(session);
-        addToObjectionsSession(session, SESSION_COMPANY_PROFILE, company);
-    }
+  const company: ObjectionCompanyProfile = await getCompanyProfile(companyNumber, token);
+
+  try {
```

Comment:
Do we need the two 'try' blocks? Could just have one block I think since error handling is the same here as above. Code then becomes more readable (focussed on the happy path).